core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/PathTrackingInputFormat.java

```diff
@@ -40,18 +48,30 @@
 public class PathTrackingInputFormat extends FileInputFormat<NullWritable, StructuredRecord> {
   private static final String PATH_FIELD = "path.tracking.path.field";
   private static final String FILENAME_ONLY = "path.tracking.filename.only";
+  private static final String FORMAT = "path.tracking.format";
+  private static final String SCHEMA = "path.tracking.inputSchema";
+  private static Job job;
 
   /**
    * Configure the input format to use the specified schema and optional path field.
    */
-  public static void configure(Configuration conf, @Nullable String pathField, boolean filenameOnly) {
+  public static void configure(Configuration conf, @Nullable String pathField, boolean filenameOnly,
+                               String format, @Nullable String schema) {
     if (pathField != null) {
       conf.set(PATH_FIELD, pathField);
     }
     conf.setBoolean(FILENAME_ONLY, filenameOnly);
+    conf.set(FORMAT, format);
+    if (schema != null) {
+      conf.set(SCHEMA, schema);
+    }
+  }
+
+  public static void setJob(Job currentJob) {
```

Comment:
yeah this is not needed because as you pointed out below, the property should be set when configuring the job. So this is useless.